tests/cypress/scripts/cliHelper.js

```diff
@@ -3,36 +3,221 @@
-  checkIfLoggedIn: () => {
-    cy.url().then((res) => {
-      if (res.includes('oauth-openshift')) {
+  /**
+   * Create new instance of the resource object within the test cluster environment.
+   * @param {object} resource The resource object to create. (Supported: application, deployment, namespace)
+   * @param {string} kubeconfig The kubeconfig file to create the resource object. (Only required for managed cluster testing)
+   * @param {object} options Additional options for creating the new resources object.
+   */
+  createResource: (resource, kubeconfig = '', options = {}) => {
+    // Build command line arguments for the resource creation.
+    var cmd = `${kubeconfig} oc get ${resource.kind} ${resource.name}`
+
+    // Check to see if the namespace was passed in the object.
+    if (resource.namespace) {
+      cmd += ` -n ${resource.namespace}`
+    }
+
+    cy.exec(cmd, { failOnNonZeroExit: false }).then((res) => {
+      if (!res.stderr) {
+        cy.log(
+          `${resource.kind}: ${resource.name} exist within the current cluster environment.`
+        )
+        Cypress.env(resource.kind, true)
+      } else {
+        cy.log(
+          `${resource.kind}: ${resource.name} does not exist within the cluster environment. Preparing to create ${resource.kind} resource.`
+        )
+        Cypress.env(resource.kind, false)
+      }
+
+      if (!Cypress.env(resource.kind)) {
+        cmd = cmd.replace('oc get', 'oc create')
+
+        // Check to see if the image was passed in the object.
+        if (resource.image) {
+          cmd += ` --image ${resource.image}`
+        }
+
+        if (resource.kind == 'application') {
+          cy.readFile('tests/cypress/templates/application.yaml').then(
+            (cfg) => {
+              let b64Cfg = btoa(
+                cfg
+                  .replaceAll('APPNAME', resource.name)
+                  .replaceAll('NAMESPACE', resource.namespace)
+              )
+              cy.exec(`echo ${b64Cfg} | base64 -d | oc apply -f -`).then(() => {
+                Cypress.env(resource.kind, true)
+              })
+            }
+          )
+        } else {
+          cy.exec(cmd).then(() => {
+            Cypress.env(resource.kind, true)
+          })
+        }
+      }
+    })
+  },
+
+  /**
+   * Delete instance of the resource object within the test cluster environment.
+   * @param {object} resource The resource object to delete.
+   * @param {string} kubeconfig The kubeconfig file to delete the resource object. (Only required for managed cluster testing)
+   * @param {object} options Additional options for deleting the resource object.
+   */
+  deleteResource: (resource, kubeconfig = '', options = {}) => {
+    cy.log(`Preparing to cleanup ${resource.kind} created during test run.`)
+
+    var resourceExist = false
+    var cmd = `${kubeconfig} oc get ${resource.kind} ${resource.name}`
+
+    if (resource.namespace) {
+      cmd += ` -n ${resource.namespace.toLowerCase()}`
```

Comment:
Why we need `.toLowerCase()` here but we don't have it in other places?

Reply:
Removing this `.toLowerCase()`. This was leftover from an earlier commit.